core/src/main/java/org/jdbi/v3/core/result/ResultIterable.java

```diff
@@ -303,4 +315,103 @@ default <U> U reduce(U identity, BiFunction<U, T, U> accumulator) {
                 });
         }
     }
+
+    /**
+     * Filters the {@link ResultIterable} by applying the specified predicate.<br>
+     * This method has the look and feel of {@link Stream#filter(Predicate)} without making use of streams.
+     *
+     * @param predicate a non-interfering, non-null predicate to apply to each element
```

Comment:
I like that you are referring to Stream documentation to help guide this, but I think the term `non-interfering` might not be quite accurate here. Since the data source is from the database, it explicitly handles concurrency (for example, Postgres would continue to provide the same snapshot data from the start of the transaction regardless of later interference).